src/formatDate.js

```diff
@@ -7,8 +7,35 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const formatedDate = [];
+  const splitedDate = date.split(fromFormat[3]);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    for (let j = 0; j < toFormat.length - 1; j++) {
+      if (fromFormat[i] === 'YYYY' && toFormat[j] === 'YY') {
+        formatedDate[j] = splitedDate[i].slice(2);
+        break;
+      }
+
+      if (fromFormat[i] === 'YY' && toFormat[j] === 'YYYY') {
+        if (splitedDate[i] < 30) {
+          formatedDate[j] = '20' + splitedDate[i];
+        } else {
+          formatedDate[j] = '19' + splitedDate[i];
+        }
+
+        break;
+      }
+
+      if (fromFormat[i] === toFormat[j]) {
+        formatedDate[j] = splitedDate[i];
+      }
+    }
+  }
+
+  return formatedDate.join(toFormat[3]);
```

Comment:
The function does not handle the reordering of the date parts. For example, if the 'fromFormat' is ['YYYY', 'MM', 'DD', '-'] and the 'toFormat' is ['DD', 'MM', 'YYYY', '/'], the function will not correctly reorder the date parts. A mapping strategy based on the 'fromFormat' and 'toFormat' arrays should be implemented to ensure correct reordering.